proposals/extensions.md

```diff
@@ -277,44 +275,37 @@ public static class Enumerable
 
 When an extension member lookup is attempted, all extension declarations within static classes that are `using`-imported contribute their members as candidates,
 regardless of receiver type. Only as part of resolution are candidates with incompatible receiver types discarded.
-A full generic type inference is attempted between the type of the actual receiver and any type parameters in the declared receiver type.
+A full generic type inference is attempted between the type of the arguments (including the actual receiver) and any type parameters (combining those in the declared receiver type and the extension member).
 
-The inferrability and uniqueness rules mean that the name of the enclosing static type is sufficient to disambiguate 
-between extension members on a given receiver type. 
-As a strawman, consider `E @ T` as a disambiguation syntax meaning on a given expression `E` begin member lookup for an immediately enclosing expression in type `T`. For instance:
+Similarly to classic extension methods, the emitted implementation methods can be invoked statically.  
+This allows passing all the type arguments explicitly, and allows the compiler to disambiguate between extension members with the same name and arity.  
```

Comment:
It would be good to illustrate a scenario where a disambiguation might be necessary, because it won't-help/not-needed for members declared within the same extension type, I think.